__tests__/app.test.js

```diff
@@ -136,3 +136,54 @@ describe("/api/articles/article:id/comments", () => {
    });
  });
 });
+describe("POST /api/articles/:article_id/comments", () => {
+ test("should respond with correct status and newly created comment", () => {
+  const testComment = { username: "lurker", body: "Wow this is incredible" };
+  return request(app)
+   .post("/api/articles/11/comments")
+   .send(testComment)
+   .expect(201)
+   .then(({ body }) => {
+    expect(body.comment).toEqual(
+     expect.objectContaining({
+      comment_id: 19,
+      body: "Wow this is incredible",
+      article_id: 11,
+      author: "lurker",
+      votes: 0,
+      created_at: expect.any(String),
+     })
+    );
+   });
+ });
+ test("should respond with an error when passed a bad request for article_id", () => {
+  const testComment = { username: "lurker", body: "Wow this is incredible" };
+  return request(app)
+   .post("/api/articles/eleven/comments")
+   .send(testComment)
+   .expect(400)
+   .then(({ body }) => {
+    expect(body.msg).toBe("bad request");
+   });
+ });
+ test("should respond with an error when passed an article_id not matching in the DB", () => {
+  const testComment = { username: "lurker", body: "Wow this is incredible" };
+  return request(app)
+   .post("/api/articles/999/comments")
+   .send(testComment)
+   .expect(400)
+   .then(({ body }) => {
+    expect(body.msg).toBe("bad request");
+   });
+ });
+ test("should respond with appropriate error when username does not exist, but valid article_id is passed", () => {
+  const testComment = { username: "bananaman", body: "Wow this is incredible" };
+  return request(app)
+   .post("/api/articles/11/comments")
+   .send(testComment)
+   .expect(404)
+   .then(({ body }) => {
+    expect(body.msg).toBe("user not found");
+   });
+ });
+});
```

Comment:
There are more error tests you should include here:
- What if one of the necessary keys ( `username` or `body` ) is missing?

And another happy path test you could include is:
- If provided with additional unnecessary keys in the request body, will your endpoint ignore them and complete the post using the necessary values it is given ?